Feature/57 allow commands scripts

sascha-andres opened the pull request:
Current state: add
Still missing: edit, remove

Refactoring: code 

Closes #57

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/sascha-andres/devenv/63)
<!-- Reviewable:end -->

Commit:
feat(#57): add command and do basic checks

Commit:
feat(#57): add command and do basic checks

Commit:
merge: branch 'feature/57-allow-commands-scripts' of github.com:sascha-andres/devenv into feature/57-allow-commands-scripts

Commit:
feat: add addscript command

#57

Commit:
refactgor: reorganize code

- moved each command into own package
- resolved resulting import cycles
- one file per function

Commit:
feat(#57): make scripts executable

on windows nothing is done

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 3c8d659520cf4cade5637e7326f26786e9f4ff99
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 22e88324646244eef80a166a12f40cb7e8d8eea1
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/sascha-andres/devenv

---
This message was auto-generated by https://gitcop.com

Commit:
feat(#57): add editscript command

works only on Linux and OSX as I assume the
EDITOR env variable to be present only there

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 3c8d659520cf4cade5637e7326f26786e9f4ff99
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 22e88324646244eef80a166a12f40cb7e8d8eea1
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/sascha-andres/devenv

---
This message was auto-generated by https://gitcop.com

Commit:
refactor: move functions to separate files

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 3c8d659520cf4cade5637e7326f26786e9f4ff99
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 22e88324646244eef80a166a12f40cb7e8d8eea1
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 67d7ef1def430d4b61c2c5529d9ee9405d6b699e
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/sascha-andres/devenv

---
This message was auto-generated by https://gitcop.com

Commit:
refactor(#57): rename command & file

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 3c8d659520cf4cade5637e7326f26786e9f4ff99
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 22e88324646244eef80a166a12f40cb7e8d8eea1
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 67d7ef1def430d4b61c2c5529d9ee9405d6b699e
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/sascha-andres/devenv

---
This message was auto-generated by https://gitcop.com

Commit:
feat(#57): add delscript command

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 3c8d659520cf4cade5637e7326f26786e9f4ff99
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 22e88324646244eef80a166a12f40cb7e8d8eea1
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 67d7ef1def430d4b61c2c5529d9ee9405d6b699e
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/sascha-andres/devenv

---
This message was auto-generated by https://gitcop.com

Commit:
refactor(57): move functions to files

add copyright header

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 3c8d659520cf4cade5637e7326f26786e9f4ff99
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 22e88324646244eef80a166a12f40cb7e8d8eea1
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 67d7ef1def430d4b61c2c5529d9ee9405d6b699e
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/sascha-andres/devenv

---
This message was auto-generated by https://gitcop.com

Commit:
feat(57): add commands to autocomplete

GitCop commented:
There were the following issues with your Pull Request

 * Commit: 3c8d659520cf4cade5637e7326f26786e9f4ff99
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 22e88324646244eef80a166a12f40cb7e8d8eea1
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: 67d7ef1def430d4b61c2c5529d9ee9405d6b699e
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/sascha-andres/devenv

---
This message was auto-generated by https://gitcop.com
